lib/user/setCookie.js

```diff
@@ -18,22 +18,20 @@ exports.optional = ['validate']
  * })
 **/
 
-exports.func = function (args) {
+exports.func = async function (args) {
   // verify it
   if (!args.cookie.toLowerCase().includes('warning:-')) {
     console.error('Warning: No Roblox warning detected in provided cookie. Ensure you include the entire .ROBLOSECURITY warning.')
   }
   if (args.validate === false) {
     options.jar.session = args.cookie
     return false
-  } else {
-    return getCurrentUser({ jar: { session: args.cookie } })
-      .then(function (res) {
-        options.jar.session = args.cookie
-        return res
-      }).catch(function (e) {
-        console.error('Failed to validate cookie: Are you sure the cookie is valid?\nEnsure you include the full cookie, including warning text.')
-        throw new Error(e)
-      })
+  }
+  try {
+    const res = await getCurrentUser({ jar: { session: args.cookie } })

```

Comment:
Could `return await getCurrentUser(...);` to remove an unnecessary line and variable.

Reply:
Or, a better solution would be to match the original behaviour and actually set the session token following validation!

Please test your PRs. Reading your code, if validate is enabled the internally stored cookie will never be updated.